### PR TITLE
Fix create issue endpoint

### PR DESCRIPTION
**Pull request by pedrorijo91**
Bug due to endpoint not accepting json values.

Problem reported in:

* https://bitbucket.org/site/master/issues/12099/error-creating-issue-through-api

Also reported, and solutioned in:

* https://bitbucket.org/site/master/issues/8620/the-repository-issues-components-not

### CODE REVIEW

**Comment by mrfyda** [comment]
does this applies to stash-scala-client as well?

**Comment by pedrorijo91** [comment]
maybe @nmatpt knows

File: src/main/scala/com/codacy/client/bitbucket/client/BitbucketClient.scala [hunk 1]
```diff
@@ -78,6 +78,14 @@ class BitbucketClient(key: String, secretKey: String, token: String, secretToken
     value
   }
 
+  def postForm[T](request: Request[T], values: Map[String, Seq[String]])(implicit reader: Reads[T]): RequestResponse[T] = {
```
Comment (johannegger) [inline src/main/scala/com/codacy/client/bitbucket/client/BitbucketClient.scala]: why would you want to implement an explicit version of post for form and json and make the more general private? the general post is imho the way to go here.

Comment (rtfpessoa) [inline src/main/scala/com/codacy/client/bitbucket/client/BitbucketClient.scala]: We do not want anybody to use the post. The api is limited to the post json or post form data.

**Comment by rtfpessoa** [comment]
LGTM :+1:

File: src/main/scala/com/codacy/client/bitbucket/client/BitbucketClient.scala (hunk 1 above)
Comment (johannegger) [inline src/main/scala/com/codacy/client/bitbucket/client/BitbucketClient.scala]: i already talked to pedro. i just don't like the names postForm and postJson. i don't mind hiding the general method but i'd really prefer if the other two were both named post with different signatures

Comment (rtfpessoa) [inline src/main/scala/com/codacy/client/bitbucket/client/BitbucketClient.scala]: The names look really good. If you look at other libraries with same
functionality they also use postForm. Play WS is not a good example in this
case. Scala http is.

I really prefer stuff to be explicit. Why leave stuff for interpretation
when you can make it explicit.

**Pull request merged**